qt_ui/windows/groundobject/QGroundObjectBuyMenu.py

```diff
@@ -0,0 +1,284 @@
+
+    def on_group_changed(self) -> None:
+        self.group_template.set_enabled(self.group_selector.isChecked())
+        if self.group_template.can_be_modified:
+            unit_type = self.unit_selector.itemData(self.unit_selector.currentIndex())
+            self.group_template.unit_count = [self.amount_selector.value()]
+            self.group_template.set_unit_type(unit_type.dcs_id)
+        self.group_template_changed.emit(self.group_template)
+
+
+class QGroundObjectTemplateLayout(QGroupBox):
+    def __init__(
+        self,
+        game: Game,
+        ground_object: TheaterGroundObject,
+        template_changed_signal: Signal(GroundObjectTemplate),
+        current_group_value: int,
+    ):
+        super(QGroundObjectTemplateLayout, self).__init__("Groups:")
+        # Connect to the signal to handle template updates
+        self.game = game
+        self.ground_object = ground_object
+        self.template_changed_signal = template_changed_signal
+        self.template_changed_signal.connect(self.load_for_template)
+        self.template: Optional[GroundObjectTemplate] = None
+
+        self.current_group_value = current_group_value
+
+        self.buy_button = QPushButton("Buy")
+        self.buy_button.clicked.connect(self.buy_group)
+
+        self.template_layout = QGridLayout()
+        self.setLayout(self.template_layout)
+
+        self.template_grid = QGridLayout()
+        self.template_layout.addLayout(self.template_grid, 0, 0, 1, 2)
+        self.template_layout.addWidget(self.buy_button, 1, 1)
+        stretch = QVBoxLayout()
+        stretch.addStretch()
+        self.template_layout.addLayout(stretch, 2, 0)
+
+    def load_for_template(self, template: GroundObjectTemplate) -> None:
+        self.template = template
+
+        # Clean the current grid
+        for id in range(self.template_grid.count()):
+            self.template_grid.itemAt(id).widget().deleteLater()
+
+        for g_id, group in enumerate(template.groups):
+            group_row = QGroundObjectGroupTemplate(g_id, group)
+            group_row.group_template_changed.connect(self.group_template_changed)
+            self.template_grid.addWidget(group_row)
+
+        self.update_price()
+
+    def group_template_changed(self, group_template: GroupTemplate) -> None:
+        self.update_price()
+
+    def update_price(self) -> None:
+        price = "$" + str(self.template.estimated_price_for(self.ground_object))
+        self.buy_button.setText(f"Buy [{price}M][-${self.current_group_value}M]")
+
+    def buy_group(self):
+        if not self.template:
+            return
+        groups = self.generate_groups()
+
+        price = 0
+        for group in groups:
+            for unit in group.units:
+                if unit.unit_type:
+                    price += unit.unit_type.price
+
+        price -= self.current_group_value
+
+        if price > self.game.blue.budget:
+            self.error_money()
+            self.close()
```

Comment:
Good question. i dont havy any clue because this is copy pasta from the original code (git is not able to handle this as this was moved out of the file).
But i will go with the disable button approach which should be quite easy to implement. I'll have to change the data-model of the UI anyway with the recent changes which made "layout" (aka template) completly generic